plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -125,7 +129,13 @@ rules plugin = do
     ideas <- if hlintOn' then getIdeas file else return (Right [])
     return (diagnostics file ideas, Just ())
 
-  getHlintSettingsRule (HlintEnabled [])
+  defineNoFile $ \GetHlintSettings -> do
+    config <- getClientConfigAction def
+    let pluginConfig = configForPlugin config plugin
+    getHlintSettingsRule $ HlintEnabled $
+      case fromJSON $ Object $ plcConfig pluginConfig of
```

Comment:
Please use Ide.Plugin.Properties to define this custom config, instead of parsing json object directly. You may want to take a look at type lenses plugin or tactics plugin to see how to use it.